### PR TITLE
Fix MSP flow, make MSP timeout dynamic

### PR DESCRIPTION
**Pull request by haslinghuis**
- [x] Add dynamic timeout for MSP calls with a save range 250- 2000ms (bounces back gradually to 4 fps).
- [x] Fix `MSP_STATUS` vs `MSP_STATUS_EX` as it was called unsupported (without software version control).
- [x] Fix [some] MSP calls stumbling over each other, without callback.
- [x] Remove checkUpdate profile from MSPHelper library as code should be in the caller.
- [x] Disconnect after crc packet error threshold has exceeded.

Note:
With a lower timeout the code shows more timeouts but overall the code is processing more calls optimizing processing.

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

File: src/js/tabs/motors.js
Comment (McGiverGim) [inline src/js/tabs/motors.js:582]: What does this change?

Comment (haslinghuis) [reply src/js/tabs/motors.js:582]: Need some more time here. Put the PR in draft. The change is not needed anymore, but was suppressing an error where the variable was not yet set,

File: src/js/serial_backend.js
Comment (chmelevskij) [inline src/js/serial_backend.js:721]: I keep seeing this pattern. But what is the value of this? Especially `Promise.resolve(true)`

I think you could achieve similar flow with using `async` on the function and then just `await`ing these calls instead.

Comment (haslinghuis) [reply src/js/serial_backend.js]: Fixed.

**Review by chmelevskij — APPROVED** [review]

**Review by limonspb — APPROVED** [review]

File: src/main.html
Comment (chmelevskij) [inline src/main.html:400]: ❤️

**Review by chmelevskij — DISMISSED** [review]

File: src/js/msp/MSPHelper.js
Comment (McGiverGim) [inline src/js/msp/MSPHelper.js:725]: If we don't need this, please remove the code, better to not left code commented. This log is only shown when modifying the motor speed manually or it is called in a timer manner?

**Review by McGiverGim — CHANGES_REQUESTED** [review]
A little change. ~And as always, any change is better if we add a screen capture with it ;)~

EDIT: Sorry, I saw the capture in the first message...

**Comment by McGiverGim** [comment]
The artifacts and the Sonar issues have disappear. I wanted to test how it appears with small window, it seems it will not fit (I'm out of my habitual computer so I can't build it at this moment).

I suppose that after fixing my little comment it will appear again and I will test it.

File: src/js/msp/MSPHelper.js
Comment (haslinghuis) [reply src/js/msp/MSPHelper.js:725]: Indeed the log is called when using motor testing.

Maybe remove the statistics as I don't think it's useful in user perspective and space is a constraint here as it does not work on small screen.

Comment (McGiverGim) [reply src/js/msp/MSPHelper.js]: It does not seem very useful, except as curiosity. But at least, we need to hide it when the window is small.

Comment (haslinghuis) [reply src/js/msp/MSPHelper.js]: Removed visual part. Also why we need configurator build, firmware target and firmware version on the taskbar?  Did not check on Android but it's already beneath the logo besides firmware info is available in cli.

**Review by limonspb — APPROVED** [review]

**Review by McGiverGim — APPROVED** [review]

**Review by blckmn — APPROVED** [review]

**Pull request merged**
